h2/src/main/org/h2/jdbc/JdbcConnection.java

```diff
@@ -1640,16 +1629,7 @@ public Blob createBlob() throws SQLException {
             int id = getNextId(TraceObject.BLOB);
             debugCodeAssign("Blob", TraceObject.BLOB, id, "createClob()");
             checkClosedForWrite();
-            try {
-                Value v = session.getDataHandler().getLobStorage().createBlob(
-                        new ByteArrayInputStream(Utils.EMPTY_BYTES), 0);
-                synchronized (session) {
-                    session.addTemporaryLob(v);
```

Comment:
I don't see where the session.addTemporaryLob code ends up in the new code?

Reply:
Allocation of a useless temporary empty `ValueLobDb` is removed from no-arg `createBlob()`, `createClob()`, and `createNClob()` methods, so all related code is removed too.

JDBC LOB classes only use `Value.getInputStream()` and `Value.getReader()`, so `ValueBytes.EMPTY` and `ValueString.EMPTY` constants are used instead. They return the same empty input stream and empty reader. In normal cases even these methods are not used, because empty value is replaced with a new BLOB or CLOB. But if it not replaced for some reason constants can provide all necessary information about empty value.

Methods with arguments that create LOBs with content are not touched.